packages/core/src/L3/nodejs/scripts/CleanScript.ts

```diff
@@ -0,0 +1,57 @@
+import { Construct } from "constructs";
+import { Bindings, LifeCycle, LifeCycleStage, Project } from "../../../L1";
+import { ManifestEntry } from "../../../L2";
+import { PackageDependency, PackageDependencyType } from "../constructs";
+
+export class CleanScript extends Construct {
+  public static readonly ID = "CleanScript";
+
+  public static hasSupport(construct: Construct) {
+    return !!this.tryOf(construct);
+  }
+
+  public static of(construct: Construct) {
+    const ret = this.tryOf(construct);
+
+    if (!ret) {
+      throw new Error(`Construct ${construct} does not have CleanScript`);
+    }
+
+    return ret;
+  }
+
+  public static tryOf(construct: Construct) {
+    return Bindings.of(Project.of(construct)).findByClass<CleanScript>(CleanScript);
+  }
+
+  private paths: Set<string> = new Set();
+
+  constructor(scope: Construct) {
+    super(scope, CleanScript.ID);
+
+    Bindings.of(Project.of(this)).bind(this);
+
+    LifeCycle.implement(this);
+    LifeCycle.of(this).on(LifeCycleStage.BEFORE_SYNTH, () => {
+      if (!this.paths.size) {
+        return;
+      }
+
+      new PackageDependency(this, "rimraf", { type: PackageDependencyType.DEV });
+
+      new ManifestEntry(this, "Scripts", {
+        scripts: {
+          clean: `rimraf ${[...this.paths].join(" ")}`,
+        },
+      });
+    });
+  }
+
+  public addPath(path: string) {
+    this.paths.add(path);
+  }
+
+  public addPaths(paths: string[]) {
+    paths.forEach((p) => this.paths.add(p));
```

Comment:
```suggestion
    paths.forEach((p) => new CleanScriptPath(this, p, p);
    
```

Reply:
This goes against the point of CleanScriptPath

it works similar to `new GitIgnore()` You call it in the construct that needs to add a path to the clean script.

Reply:
But `GitIgnore` extends `AppendFile`, which implements a lot of stuff. Right now I don't think it is at all the same as GitIgnore?

Reply:
Yep but all the added files are stored in the root class as fields. I'm doing the same, storing all paths in the CleanScript singleton. Seems more efficient to me, then instead having to loop over all CleanScript instances on synth.

Reply:
But in my mind the point, and value of, OOP is isolation. And one the main points of `constructs` is that it is a graph, which you can traverse, inspect and amend.

Hypothetical example.

Some package adds a clean script path, but I, as a consumer, want to actually remove it.

What I would expect is to iterate all `CleanScriptPath` objects, and then remove the object from the scope.

This would intern remove the path from the clean script parameters.

But now, what CleanScriptPath does is simply proxy the request down to the CleanScript class and that's it.

Which means operating on the CleanScriptPath object has no effect.

Imagine the scenario:

```
// change the path
cleanScriptPath.change('some/other/path')
```

Reply:
A great example in CDK-land is Aspects.

You can do:

```
Aspects.of(project).add({
  visit(node) {
	// find any class that is extended from Construct
    if (node instanceof SomeConstructChildClass) {
      node.foo()
    }
  }
})
```